app/index.tsx

```diff
@@ -78,163 +78,164 @@ const parseDeepLinking = (url: string) => {
+			initCrashReport();
 		}
-		const { width, height, scale, fontScale } = Dimensions.get('window');
-		const theme = initialTheme();
-		this.state = {
-			theme: getTheme(theme),
-			themePreferences: theme,
-			width,
-			height,
-			scale,
-			fontScale
-		};
+
 		if (isTablet) {
-			this.initTablet();
+			initTablet();
 		}
-		setNativeTheme(theme);
-	}
 
-	componentDidMount() {
-		this.listenerTimeout = setTimeout(() => {
+		// Set the native theme for the app
+		setNativeTheme(initTheme);
+
+		const timeout = setTimeout(() => {
 			Linking.addEventListener('url', ({ url }) => {
 				const parsedDeepLinkingURL = parseDeepLinking(url);
 				if (parsedDeepLinkingURL) {
 					store.dispatch(deepLinkingOpen(parsedDeepLinkingURL));
 				}
 			});
 		}, 5000);
-		this.dimensionsListener = Dimensions.addEventListener('change', this.onDimensionsChange);
-	}
-
-	componentWillUnmount() {
-		clearTimeout(this.listenerTimeout);
-		this.dimensionsListener?.remove?.();
-
-		unsubscribeTheme();
-	}
 
-	init = async () => {
-		store.dispatch(appInitLocalSettings());
+		const dimensionsListener: EmitterSubscription = Dimensions.addEventListener('change', onDimensionsChange);
 
-		// Open app from push notification
-		const notification = await initializePushNotifications();
-		if (notification) {
-			onNotification(notification);
-			return;
-		}
-
-		await getInitialNotification();
-
-		// Open app from deep linking
-		const deepLinking = await Linking.getInitialURL();
-		const parsedDeepLinkingURL = parseDeepLinking(deepLinking!);
-		if (parsedDeepLinkingURL) {
-			store.dispatch(deepLinkingOpen(parsedDeepLinkingURL));
-			return;
-		}
-
-		// Open app from app icon
-		store.dispatch(appInit());
-	};
+		return () => {
+			clearTimeout(timeout);
+			dimensionsListener.remove();
+			unsubscribeTheme();
+		};
+	}, []);
 
-	getMasterDetail = (width: number) => {
-		if (!isTablet) {
-			return false;
-		}
-		return width > MIN_WIDTH_MASTER_DETAIL_LAYOUT;
-	};
+	useEffect(() => {
+		// Subscribe to Appearance changes whenever themePreferences changes
+		subscribeTheme(themePreferences, () => setTheme(themePreferences));
+	}, [themeState.themePreferences]); // Dependency array ensures this runs when `themePreferences` changes
 
-	setMasterDetail = (width: number) => {
-		const isMasterDetail = this.getMasterDetail(width);
-		store.dispatch(setMasterDetailAction(isMasterDetail));
+	const initCrashReport = () => {
+		getAllowCrashReport().then(allowCrashReport => {
+			toggleCrashErrorsReport(allowCrashReport);
+		});
+		getAllowAnalyticsEvents().then(allowAnalyticsEvents => {
+			toggleAnalyticsEventsReport(allowAnalyticsEvents);
+		});
 	};
 
-	// Dimensions update fires twice
-	onDimensionsChange = debounce(({ window: { width, height, scale, fontScale } }: { window: IDimensions }) => {
-		this.setDimensions({
+	// Handle dimensions change
+	const onDimensionsChange = debounce(({ window: { width, height, scale, fontScale } }: { window: IDimensions }) => {
```

Comment:
This kind of debounce won't work on function components.
You should use useDebounce or useDebounceHook